src/App.tsx

```diff
@@ -1,14 +1,45 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useContext, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { getUser } from './api';
+import { TodosContext } from './components/TodosProvider';
 
 export const App: React.FC = () => {
+  const [userData, setUserData] = useState<User | null>(null);
+  const [currentTodo, setCurrentTodo] = useState<Todo | null>(null);
+  const [isModalLoading, setIsModalLoading] = useState(false);
+
+  const { todos, loadTodosError } = useContext(TodosContext);
+
+  const getDataToModal = useCallback(async (userId: number, selectedTodo: Todo) => {
+    try {
+      setIsModalLoading(true);
+      const selectedUser = await getUser(userId);
+
+      setCurrentTodo(selectedTodo);
```

Comment:
don't repeat yourself: you repeated this code for both of cases, in `try` and in `catch` block, let's think how we can optimize it